src/NUnitFramework/framework/Constraints/PropertyConstraint.cs

```diff
@@ -70,7 +70,8 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
                 throw new ArgumentException(string.Format("Property {0} was not found", name), "name");
 
             propValue = property.GetValue(actual, null);
-            return new ConstraintResult(this, propValue, BaseConstraint.ApplyTo(propValue).IsSuccess);
+            var baseResult = BaseConstraint.ApplyTo(propValue);
+            return new ConstraintResult(this, baseResult.ActualValue, baseResult.IsSuccess);
```

Comment:
I'm in favor of the minimal fix here. When I first read the chain of constraints in the issue, it took me awhile to parse it. To me, even though it is a valid construct, it isn't easy to understand, so that would lower my priority on fixing it. If you create overly complex asserts, you should expect error messages of the same complexity 😄 I tend to break things down into multiple asserts for that reason.